cmd/socat/main.go

```diff
@@ -0,0 +1,143 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"net"
+	"os"
+	"strings"
+
+	"github.com/spf13/cobra"
+)
+
+func main() {
+	cmd := newRootCmd()
+	err := cmd.Execute()
+	if err != nil {
+		os.Exit(1)
+	}
+}
+
+func newRootCmd() *cobra.Command {
+	var uniDir bool
+	cmd := cobra.Command{
+		Use:   "socat <address> <address>",
+		Short: "Minimalistic socat.",
+		Long: `Minimalistic socat.
+Implements only TCP, OPEN and stdio ("-") addresses with no options.
+Only supported flag is -u.`,
+		Args: cobra.ExactArgs(2),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			right, err := createConnection(args[0])
```

Comment:
Minor nit, but if this is arg 0, and the first argument to `connect` later; perhaps this should be called `left` and the other `right`.

Reply:
my bad